lua/entities/gmod_tardis/modules/sh_health.lua

```diff
@@ -0,0 +1,174 @@
+})
+
+TARDIS:AddSetting({
+    id="health-max",
+    name="Max Health",
+    desc="Maximum ammount of health the TARDIS has",
+    section="Misc",
+    type="number",
+    value=1000,
+    min=1,
+    max=50000,
+    networked=true
+})
+
+ENT:AddHook("Initialize","health-init",function(self)
+    --local max_health = self:GetSetting("health-max") or 100
+    self:SetData("health-val", TARDIS:GetSetting("health-max"), true)
+end)
+function ENT:ChangeHealth(NewHealth)
+    if not TARDIS:GetSetting("health-enabled") then return end
+    if self:GetData("repairing", false) then
+        return
+    end
+    local OldHealth = self:GetHealth()
+    if NewHealth <= 0 then 
+        NewHealth = 0
+        if NewHealth == 0 and not ( NewHealth == OldHealth) then
+            self:CallHook("health-depleted")
+            self.interior:CallHook("health-depleted")
+        end
+    end
+    self:SetData("health-val", NewHealth, true)
+    self:CallHook("health-change")
+end
+
+function ENT:GetHealth()
+    return self:GetData("health-val", 0)
+end
+
+function ENT:GetRepairTime()
+    return self:GetData("repair-time")-CurTime()
+end
+
+if SERVER then
+    cvars.AddChangeCallback("tardisrw_maxhealth", function(cvname, oldvalue, newvalue)
+       TARDIS:SetSetting("health-max", tonumber(newvalue), true)
+    end, "UpdateOnChange")  
+    
+    cvars.AddChangeCallback("tardisrw_damage", function(cvname, oldvalue, newvalue)
+       TARDIS:SetSetting("health-enabled", tobool(newvalue), true)
+    end, "UpdateOnChange")
+
+    function ENT:Explode()
+        if not vFireInstalled then
+            local explode = ents.Create("env_explosion")
+            explode:SetPos( self:LocalToWorld(Vector(0,0,50)) ) //Puts the explosion where you are aiming
+            explode:SetOwner( self ) //Sets the owner of the explosion
+            explode:Spawn()
+            explode:SetKeyValue("iMagnitude","100") //Sets the magnitude of the explosion
+            explode:Fire("Explode", 0, 0 ) //Tells the explode entity to explode
+            //explode:EmitSound("weapon_AWP.Single", 400, 400 ) //Adds sound to the explosion
+        else
+            //alternate explosion
+        end
+    end
+
+    function ENT:ToggleRepair()
+        local on = not self:GetData("repair-primed",false)
+        self:SetRepair(on)
+    end
+    function ENT:SetRepair(on)
+        //self.interior:SetData("selfrepair-primed",on,true)
+        if (self:GetHealth() > TARDIS:GetSetting("health-max",1)-1) or self:GetData("vortex",false) then return end
+        if on==true then
+            for k,_ in pairs(self.occupants) do
+                k:ChatPrint("This TARDIS has been set to self-repair. Please vacate the interior.")
+            end
+            if self.interior:GetData("power-state") then self.interior:SetPower(!on) end
+            self:SetData("repair-primed",on,true)
+        else
+            self:SetData("repair-primed",on,true)
+            self.interior:SetPower(!on)
```

Comment:
```suggestion
            self.interior:SetPower(true)
```
garry-ism + we already know on is false here so might as well be explicit